### ISSUE TITLE
Cut-out Pipeline: Technical Challenges & Edge Case Issues

### ISSUE THREAD

**Issue by kparikh-tenup** [ZhengPeng7/BiRefNet#315]
Hi @ZhengPeng7,

We are building an automatic cut-out pipeline for dealership vehicle photos. The output is an RGBA cut-out of the vehicle including its attachments — antennas, ladder racks, roof tubes / carriers and spoilers. Two categories of input are still failing:

* **Thin / complex structures** — antenna, rack, tube and spoiler are under-detected or have soft, incomplete edges.  
* **Partially-camouflaged vehicles** — where part of the body or an accessory is low-contrast against the background and gets merged into it, so it is not cut cleanly.

## **Issue 1 — Partially-camouflaged bodies blend into the background  (primary concern)**

When part of the vehicle is low-contrast against what is behind it, that part is not cut cleanly, and none of the models above handle it. This is our biggest blocker — the production model has ample training data for these cases yet still fails.

<img width="543" height="410" alt="Image" src="https://github.com/user-attachments/assets/47a51c8f-7dfc-4614-9ba1-51d0fc521bf6" />

## **Issue 2 — Antenna not detected reliably**

Thin vertical antennas are frequently missed or broken up, even in Experiments 1 & 2 which specifically targeted them. Their pixel area is tiny, so the segmentation losses (BCE / IoU) barely penalise dropping them.

## **Issue 3 — Racks, tubes, carriers and spoilers not crisp / not cut**

Roof ladder racks, headache-rack tubes, roof carriers and rear wings / spoilers are partially missed or have soft, incomplete edges. 

<img width="561" height="425" alt="Image" src="https://github.com/user-attachments/assets/0dc6ac4c-a640-4887-93c6-b6e1c0bc4422" />

<img width="629" height="479" alt="Image" src="https://github.com/user-attachments/assets/4a19c492-18ba-4367-b447-018577f5bd22" />

Antenna not detected properly

# **What we need help with**

1. Thin structures (antenna, tube): what is the recommended way to reliably capture them — loss re-weighting for thin / low-area regions, higher-resolution training & inference, or a trimap \+ matting refinement stage (e.g. ViTMatte / BiRefNet\_HR-matting)?  
2. Partial camouflage: which prior / architecture handles low-contrast body-vs-background best (DIS vs HR-general vs a matting head)? Is this a data problem, a loss problem, or a resolution problem?  
3. Data & annotation: does mask resolution (we are moving to ≥2048px), oversampling of thin-structure images, and a consistent see-through-glass convention meaningfully help — or are we already past the point of diminishing returns at \~950 epochs?  
4. Inference tricks: would tiled / zoom-refine inference on the rack & antenna regions be worthwhile?  
5. Anything else we should try to reach production-quality edges on these three cases.
